modules/apps/oauth2-provider/oauth2-provider-web/src/main/java/com/liferay/oauth2/provider/web/internal/AssignableScopes.java

```diff
@@ -117,14 +115,17 @@ public boolean equals(Object object) {
 	public AssignableScopes getApplicationAssignableScopes(
 		String applicationName) {
 
-		Stream<LiferayOAuth2Scope> stream = _liferayOAuth2Scopes.stream();
+		Set<LiferayOAuth2Scope> liferayOAuth2Scopes = new HashSet<>();
+
+		for (LiferayOAuth2Scope liferayOAuth2Scope : _liferayOAuth2Scopes) {
+			if (!Objects.equals(
+					applicationName, liferayOAuth2Scope.getApplicationName())) {
 
-		Set<LiferayOAuth2Scope> liferayOAuth2Scopes = stream.filter(
-			liferayOAuth2Scope -> applicationName.equals(
-				liferayOAuth2Scope.getApplicationName())
-		).collect(
-			Collectors.toSet()
-		);
+				continue;
```

Comment:
@W2jx No need to shortcut for such `for` loops. We usually use shortcut in loop to buy space, to avoid too many indents; but if the original `if` contains only 1 line and it's very short, no need to do that.